scylla/src/transport/session.rs

```diff
@@ -1151,6 +1153,34 @@ impl Session {
         }
     }
 
+    fn should_consider_query_for_latency_measurements(
+        &self,
+        result: &Result<impl AllowedRunQueryResTType, QueryError>,
+    ) -> bool {
+        self.load_balancer.requires_latency_measurements()
+            && match result {
+                Ok(_) => true,
+                Err(err) => match err {
+                    // "fast" errors, i.e. ones that are returned quickly after the query begins
+                    QueryError::BadQuery(_)
+                    | QueryError::TooManyOrphanedStreamIds(_)
+                    | QueryError::UnableToAllocStreamId
+                    | QueryError::DbError(DbError::IsBootstrapping, _)
+                    | QueryError::DbError(DbError::Unavailable { .. }, _)
+                    | QueryError::DbError(DbError::Unprepared { .. }, _)
+                    | QueryError::DbError(DbError::Overloaded { .. }, _) => false,
```

Comment:
That's an interesting problem. `Overloaded` error actually means "coordinator is overloaded" in Cassandra terminology, and in fact we'd receive this error rather quickly, because from the database server perspective, the sooner we disqualify the request, the better, because then we don't waste resources on a query that's doomed to fail. So, on the one hand, it's deceptive to count an `Overloaded` response into the latency tracker, because this response may come very fast, even though the coordinator is indeed overloaded. Drivers should be as cooperative as possible, so perhaps we should have a special case for `Overloaded` error, and induce fake high latency to the tracker (e.g. 10s), to make it less likely to pick this node in the near future. @piodul I'd appreciate your take on this too

Reply:
I agree that it doesn't make sense to count the actual latency of a request that resulted in `Overloaded` response.

I'm not sure whether I completely understand the behavior of the policy, so I can't tell if 10 seconds in particular (or any other constant) will be all right for all cases. Instead, in case we get the `Overloaded` response, we can set the node's latency to `max(current_latency, 2 * min_latency)` so that the node is naturally retried after the configured `retry_period` but the latency doesn't grow too much and there isn't too much catching up needed for the moving average to do.